Objects/listobject.c

```diff
@@ -2886,6 +2891,35 @@ listiter_setstate(listiterobject *it, PyObject *state)
     Py_RETURN_NONE;
 }
 
+static int
+listiter_contains(listiterobject *it, PyObject *el)
+{
+    PyListObject *seq;
+    PyObject *item;
+    int cmp;
+
+    assert(it != NULL);
+    seq = it->it_seq;
+    if (seq == NULL)
+        return 0;
+    assert(PyList_Check(it->it_seq));
+
+    do {
+        item = PyList_GET_ITEM(seq, it->it_index);
+        ++it->it_index;
+        if (it->it_index >= Py_SIZE(seq)) {
+            it->it_seq = NULL;
+            cmp = PyObject_RichCompareBool(el, item, Py_EQ);
+            Py_DECREF(seq);
+            return cmp;
+        }
+        cmp = PyObject_RichCompareBool(el, item, Py_EQ);
```

Comment:
If `Py_INCREF(item)` is needed here, then it is needed in `list_contains()`. But I can't reproduce any issue with it. It seems to me that `Py_INCREF(item)` is not needed.